mod/schematron/src/main/com/thaiopensource/validate/schematron/SilencingErrorListener.java

```diff
@@ -0,0 +1,20 @@
+package com.thaiopensource.validate.schematron;
+
+import net.sf.saxon.lib.StandardErrorListener;
+import net.sf.saxon.trans.XPathException;
+
+import javax.xml.transform.TransformerException;
+
+public class SilencingErrorListener extends StandardErrorListener {
+  @Override
+  public void warning(TransformerException exception)
+      throws TransformerException {
+    if (exception instanceof XPathException) {
+        XPathException xe = (XPathException) exception;
+        if ("SXWN9000".equals(xe.getErrorCodeLocalPart())) {
+            return;
```

Comment:
> This could potentially silence other warnings (the error code is not specific to this particular issue).…

Right, but if we wanted to isolate just the *“The parent axis starting at a document node will never select anything”* case, then as far as I can see the only alternative would be to do string-matching against the message text — which would only work when the messages are emitted in English, but not for other locales. Right?

Reply:
Yes, this is my understanding as well.
I made that comment for the sake of clarity, but again I think silencing all SXWN9000 is safe.